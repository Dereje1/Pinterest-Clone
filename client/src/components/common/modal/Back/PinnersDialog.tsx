import * as React from 'react';
import List from '@mui/material/List';
import Dialog from '@mui/material/Dialog';
import ListSubheader from '@mui/material/ListSubheader';
import ListItemText from '@mui/material/ListItemText';
import { getProviderIcons, UserListItem } from '../../common';
import { providerIconsType, PinnerType } from '../../../../interfaces';

const providerIcons: providerIconsType = getProviderIcons({ fontSize: 20 });

interface PinnersDialogProps {
  onCloseDialog: () => void
  onClosePin: (_: React.SyntheticEvent, forceClose?: boolean) => void
  open: boolean
  pinnersList: PinnerType[]
  displayLogin: () => void
  authenticated: boolean
}

function PinnersDialog({
  onCloseDialog,
  onClosePin,
  open,
  pinnersList,
  authenticated,
  displayLogin,
}:PinnersDialogProps) {
  const handleCose = (e: React.SyntheticEvent) => {
    onCloseDialog();
    onClosePin(e);
  };

  return (
    <Dialog
      onClose={onCloseDialog}
      open={open}
      PaperProps={{ sx: { position: 'fixed', top: 0 } }}
      disableAutoFocus
    >
      <List
        sx={{ pt: 0 }}
        subheader={<ListSubheader sx={{ textAlign: 'center' }}>Pinners</ListSubheader>}
      >
        {pinnersList.map((pinner) => (
          <UserListItem
            profileLinkProps={{
              authenticated,
              userId: pinner.userId,
              title: <ListItemText
                primary={pinner.name}
                primaryTypographyProps={{ color: '#3752ff', fontWeight: 'bold' }}
              />,
              closePin: handleCose,
              displayLogin,
            }}
            providerIcons={providerIcons}
            key={pinner.userId}
            service={pinner.service}
          />
        ))}
      </List>
    </Dialog>
  );
}

export default PinnersDialog;
