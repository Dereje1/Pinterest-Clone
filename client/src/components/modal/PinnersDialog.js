import * as React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Avatar from '@mui/material/Avatar';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import ListItemAvatar from '@mui/material/ListItemAvatar';
import ListItemText from '@mui/material/ListItemText';
import Dialog from '@mui/material/Dialog';
import { getProviderIcons } from '../common/common';

const providerIcons = getProviderIcons({ fontSize: 20 });

const PinnersDialog = ({
  onClose, open, pinnersList,
}) => (
  <Dialog
    onClose={onClose}
    open={open}
    PaperProps={{ sx: { position: 'fixed', top: 0 } }}
  >
    <List sx={{ pt: 0 }}>
      {pinnersList.map(pinner => (
        <ListItem key={pinner.userId} component={Link} to={`/profile/${pinner.userId}-${pinner.service}-${pinner.name}`}>
          <ListItemAvatar>
            <Avatar sx={{ width: 26, height: 26, bgcolor: providerIcons[pinner.service].color }}>
              {providerIcons[pinner.service].icon}
            </Avatar>
          </ListItemAvatar>
          <ListItemText primary={pinner.name} />
        </ListItem>
      ))}
    </List>
  </Dialog>
);

PinnersDialog.propTypes = {
  onClose: PropTypes.func.isRequired,
  open: PropTypes.bool.isRequired,
  pinnersList: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.string)).isRequired,
};

export default PinnersDialog;
