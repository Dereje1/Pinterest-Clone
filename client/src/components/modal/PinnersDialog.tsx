import * as React from 'react';
import Avatar from '@mui/material/Avatar';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import ListItemAvatar from '@mui/material/ListItemAvatar';
import ListItemText from '@mui/material/ListItemText';
import Dialog from '@mui/material/Dialog';
import ListSubheader from '@mui/material/ListSubheader';
import { getProviderIcons } from '../common/common';
import ProfileLink from './ProfileLink';
import { providerIconsType, PinnerType } from '../../interfaces';

const providerIcons: providerIconsType = getProviderIcons({ fontSize: 20 });

interface PinnersDialogProps {
  onCloseDialog: () => void
  onClosePin: (_: React.SyntheticEvent, forceClose?: boolean) => void
  open: boolean
  pinnersList: PinnerType[]
  displayLogin: () => void
  authenticated: boolean
}

function PinnersDialog({
  onCloseDialog,
  onClosePin,
  open,
  pinnersList,
  authenticated,
  displayLogin,
}:PinnersDialogProps) {
  const handleCose = (e: React.SyntheticEvent) => {
    onCloseDialog();
    onClosePin(e);
  };

  return (
    <Dialog
      onClose={onCloseDialog}
      open={open}
      PaperProps={{ sx: { position: 'fixed', top: 0 } }}
      disableAutoFocus
    >
      <List
        sx={{ pt: 0 }}
        subheader={<ListSubheader sx={{ textAlign: 'center' }}>Pinners</ListSubheader>}
      >
        {pinnersList.map((pinner) => (
          <ListItem
            key={pinner.userId}
          >
            <ListItemAvatar>
              <Avatar
                sx={{
                  width: 26,
                  height: 26,
                  bgcolor: providerIcons[pinner.service as keyof providerIconsType].color,
                }}
              >
                {providerIcons[pinner.service as keyof providerIconsType].icon}
              </Avatar>
            </ListItemAvatar>
            <ProfileLink
              authenticated={authenticated}
              closePin={handleCose}
              userId={pinner.userId}
              title={<ListItemText primary={pinner.name} />}
              displayLogin={displayLogin}
            />
          </ListItem>
        ))}
      </List>
    </Dialog>
  );
}

export default PinnersDialog;
