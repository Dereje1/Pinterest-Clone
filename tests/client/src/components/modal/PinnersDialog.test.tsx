import React from 'react';
import { EnzymePropSelector, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PinnersDialog from '../../../../../client/src/components/modal/PinnersDialog';
import { pinsStub } from '../../../stub';

test('The pinners dialog will render', () => {
  const wrapper = shallow(<PinnersDialog
    pinnersList={pinsStub[0].savedBy}
    open
    onCloseDialog={jest.fn()}
    onClosePin={jest.fn()}
    displayLogin={jest.fn()}
    authenticated={false}
  />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

test('will close the dialog and the pin on a profile link click', () => {
  const onCloseDialog = jest.fn();
  const onClosePin = jest.fn();
  const wrapper = shallow(<PinnersDialog
    pinnersList={pinsStub[0].savedBy}
    open
    onCloseDialog={onCloseDialog}
    onClosePin={onClosePin}
    displayLogin={jest.fn()}
    authenticated={false}
  />);
  const pinner: EnzymePropSelector = wrapper.find('ProfileLink').at(0);
  pinner.props().closePin();
  expect(onCloseDialog).toHaveBeenCalled();
  expect(onClosePin).toHaveBeenCalled();
});
