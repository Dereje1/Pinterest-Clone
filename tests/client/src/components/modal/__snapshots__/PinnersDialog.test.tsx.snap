// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The pinners dialog will render 1`] = `
<ForwardRef(Dialog)
  PaperProps={
    {
      "sx": {
        "position": "fixed",
        "top": 0,
      },
    }
  }
  disableAutoFocus={true}
  onClose={[MockFunction]}
  open={true}
>
  <ForwardRef(List)
    subheader={
      <ForwardRef(ListSubheader)
        sx={
          {
            "textAlign": "center",
          }
        }
      >
        Pinners
      </ForwardRef(ListSubheader)>
    }
    sx={
      {
        "pt": 0,
      }
    }
  >
    <UserListItem
      key="1"
      profileLinkProps={
        {
          "authenticated": false,
          "closePin": [Function],
          "displayLogin": [MockFunction],
          "title": <ForwardRef(ListItemText)
            primary="savedBy - id-1"
            primaryTypographyProps={
              {
                "color": "#3752ff",
                "fontWeight": "bold",
              }
            }
          />,
          "userId": "1",
        }
      }
      providerIcons={
        {
          "github": {
            "color": "#1d7b20",
            "icon": <Memo
              style={
                {
                  "fontSize": 20,
                }
              }
            />,
          },
          "google": {
            "color": "#4285F4",
            "icon": <Memo
              style={
                {
                  "fontSize": 20,
                }
              }
            />,
          },
          "twitter": {
            "color": "#1DA1F2",
            "icon": <Memo
              style={
                {
                  "fontSize": 20,
                }
              }
            />,
          },
        }
      }
      service="google"
    />
    <UserListItem
      key="1"
      profileLinkProps={
        {
          "authenticated": false,
          "closePin": [Function],
          "displayLogin": [MockFunction],
          "title": <ForwardRef(ListItemText)
            primary="savedBy - id-1"
            primaryTypographyProps={
              {
                "color": "#3752ff",
                "fontWeight": "bold",
              }
            }
          />,
          "userId": "1",
        }
      }
      providerIcons={
        {
          "github": {
            "color": "#1d7b20",
            "icon": <Memo
              style={
                {
                  "fontSize": 20,
                }
              }
            />,
          },
          "google": {
            "color": "#4285F4",
            "icon": <Memo
              style={
                {
                  "fontSize": 20,
                }
              }
            />,
          },
          "twitter": {
            "color": "#1DA1F2",
            "icon": <Memo
              style={
                {
                  "fontSize": 20,
                }
              }
            />,
          },
        }
      }
      service="github"
    />
  </ForwardRef(List)>
</ForwardRef(Dialog)>
`;
